xla/service/gpu/fusions/dynamic_slice_fusion_test.cc

```diff
@@ -3190,6 +3190,61 @@ TEST_F(DynamicSliceFusionTest, ReduceScatterDynamicSlice) {
                                                 false, true, error));
 }
 
+TEST_F(DynamicSliceFusionTest, OffsetComputation) {
+  const char* hlo_opt = R"(
+    HloModule test, replica_count=2
+    add {
+      a = s32[] parameter(0)
+      b = s32[] parameter(1)
+      ROOT add = s32[] add(a,b)
+    }
+    dynamic-slice-fusion {
+      src = s32[32,32] parameter(0)
+      dest = s32[32,32] parameter(1)
+      offset1 = s32[] parameter(2)
+      offset2 = s32[] parameter(3)
+      rs = s32[16,32] reduce-scatter(src), dimensions={0}, replica_groups={{0,1}}, to_apply=add
+      ROOT dus = s32[32,32] dynamic-update-slice(dest, rs, offset1, offset2)
+    }
+    ENTRY main {
+      src = s32[32,32] parameter(0)
+      dest = s32[32,32] parameter(1)
+      c0 = s32[] constant(0)
+      c10 = s32[] constant(10)
+      add = s32[] add(c0, c10)
+      ROOT fusion = s32[32,32] fusion(src, dest, c0, add), kind=kCustom, calls=dynamic-slice-fusion,
+        backend_config={"fusion_backend_config":{"kind":"__custom_fusion","custom_fusion_config":{"name":"dynamic_address_computation"}}}
+    }
+  )";
+
+  const char* hlo_ref = R"(
+    HloModule test, replica_count=2
+    add {
+      a = s32[] parameter(0)
+      b = s32[] parameter(1)
+      ROOT add = s32[] add(a,b)
+    }
+    ENTRY main {
+      src = s32[32,32] parameter(0)
+      dest = s32[32,32] parameter(1)
+      c0 = s32[] constant(0)
+      c5 = s32[] constant(5)
+      add = s32[] add(c5, c5)
+      rs.1 = ((s32[32,32]), s32[16,32]) reduce-scatter-start(src), dimensions={0}, replica_groups={{0,1}}, to_apply=add
+      rs = s32[16,32] reduce-scatter-done(rs.1)
+      ROOT dus = s32[32,32] dynamic-update-slice(dest, rs, c0, add)
+    }
+  )";
+
+  TF_ASSERT_OK_AND_ASSIGN(std::unique_ptr<HloModule> module_ref,
+                          ParseAndReturnVerifiedModule(hlo_ref));
+  TF_ASSERT_OK_AND_ASSIGN(std::unique_ptr<HloModule> module_opt,
+                          ParseAndReturnVerifiedModule(hlo_opt));
+  ErrorSpec error{1e-3, 1e-3};
+  EXPECT_TRUE(RunAndCompareTwoModulesReplicated(
+      std::move(module_ref), std::move(module_opt), false, true, error));
```

Comment:
Please use `/*arg_name=*/` prefixes when using literal values in function calls---otherwise it's hard to remember what things are supposed to be.